hydra-main/src/main/java/com/addthis/hydra/query/aggregate/MeshSourceAggregator.java

```diff
@@ -133,12 +133,36 @@ void stopSources(String message) {
         }
     }
 
-    boolean tryActivateSource(QueryTaskSourceOption option) {
-        return option.tryActivate(meshy, queryOptions);
+    /**
+     * Attemps to active one of the options of the given QueryTaskSource.
+     *
+     * @param taskSource the QueryTaskSource to activate
+     * @return {@code true} if one of the options is activated; {@code false} if none of the
+     *         options can be/is activated
+     */
+    boolean tryActivateSource(QueryTaskSource taskSource) {
+        for (QueryTaskSourceOption option : taskSource.options) {
+            if (option.tryActivate(meshy, queryOptions)) {
+                return true;
+            }
+        }
+        return false;
     }
 
-    void replaceQuerySource(QueryTaskSource taskSource, QueryTaskSourceOption option) throws Exception {
+    /**
+     * Resets a QueryTaskSource and attempts to replace the previously selected option with a new one.
+     *
+     * @param taskSource
+     * @throws Exception
+     * @return {@code true} if a new option is activated; {@code false} if selected option is {@code null} or the new
+     *         option could not be activated.
+     */
+    boolean replaceQuerySource(QueryTaskSource taskSource) throws Exception {
+        QueryTaskSourceOption option = taskSource.getSelectedSource();
         taskSource.reset();
+        if (option == null) {
```

Comment:
why are we now allowing the option to be null here? That seems like an invalid code path that we would rather have made visible by the subsequent NPE. Now it gives the illusion of being handled correctly with this pretend return value that isn't even checked anywhere.